extension3/src/main/java/eu/maveniverse/maven/nisse/extension3/internal/NisseLifecycleParticipant.java

```diff
@@ -50,4 +50,13 @@ public void afterProjectsRead(MavenSession session) throws MavenExecutionExcepti
             throw new MavenExecutionException("Nisse failed to inline", e);
         }
     }
+
+    @Override
+    public void afterSessionEnd(MavenSession session) throws MavenExecutionException {
+        try {
+            inliner.cleanup(session, session.getProjects());
```

Comment:
@lfvjimisola as you see, "after session end" (maven core callback) the POM file is cleaned up.